src/main/java/icbm/classic/lib/network/packet/PacketSpawnAirParticle.java

```diff
@@ -0,0 +1,127 @@
+    public float size;
+    /** ID of the entity controller of the blast */
+    public int entityID;
+    private int dimId;
+    private double posX;
+    private double posY;
+    private double posZ;
+    private double v;
+    private double v1;
+    private double v2;
+    private float red;
+    private float green;
+    private float blue;
+    private float scale;
+    private int ticksToLive;
+
+    public PacketSpawnAirParticle()
+    {
+        //Needed for forge to construct the packet
+    }
+
+    public PacketSpawnAirParticle(int dimId, double posX, double posY, double posZ, double v, double v1, double v2,
+                                  float red, float green, float blue, float scale, int ticksToLive)
+    {
+
+        this.dimId = dimId;
+        this.posX = posX;
+        this.posY = posY;
+        this.posZ = posZ;
+        this.v = v;
+        this.v1 = v1;
+        this.v2 = v2;
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
+        this.scale = scale;
+        this.ticksToLive = ticksToLive;
+    }
+
+    @Override
+    public void encodeInto(ChannelHandlerContext ctx, ByteBuf buffer)
+    {
+        /*buffer.writeFloat(size);
+        ByteBufUtils.writeVarInt(buffer, entityID, 5);*/
+        buffer.writeInt(dimId);
+        buffer.writeDouble(posX);
+        buffer.writeDouble(posY);
+        buffer.writeDouble(posZ);
+        buffer.writeDouble(v);
+        buffer.writeDouble(v1);
+        buffer.writeDouble(v2);
+        buffer.writeFloat(red);
+        buffer.writeFloat(green);
+        buffer.writeFloat(blue);
+        buffer.writeFloat(scale);
+        buffer.writeInt(ticksToLive);
+    }
+
+    @Override
+    public void decodeInto(ChannelHandlerContext ctx, ByteBuf buffer)
+    {
+       /* size = buffer.readFloat();
+        entityID = ByteBufUtils.readVarInt(buffer, 5);*/
+        dimId=buffer.readInt();
+        posX=buffer.readDouble();
+        posY=buffer.readDouble();
+        posZ=buffer.readDouble();
+        v=buffer.readDouble();
+        v1=buffer.readDouble();
+        v2=buffer.readDouble();
+        red=buffer.readFloat();
+        green=buffer.readFloat();
+        blue=buffer.readFloat();
+        scale=buffer.readFloat();
+        ticksToLive=buffer.readInt();
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public void handleClientSide(EntityPlayer player)
+    {
+        if (Minecraft.getMinecraft().world != null && player.world.provider.getDimension() == dimId)
+        {
+            /*final Entity entity = Minecraft.getMinecraft().world.getEntityByID(entityID);
+            if (entity instanceof EntityExplosion && ((EntityExplosion) entity).getBlast() instanceof BlastRedmatter)
+            {
+                ((BlastRedmatter) ((EntityExplosion) entity).getBlast()).size = size;
+            }*/
+
+            ICBMClassic.proxy.spawnAirParticle(player.world, new Pos(posX,posY,posZ), v, v1, v2, red, green ,blue, scale, ticksToLive );
```

Comment:
Need a config to wrap around this to disable the particle spawning. You don't have to do that in this PR but we should create an issue as a reminder.